facebook/ThreatExchange
# Refactor pdq indexer

Summary
---------

The way pdq indexer reads all PDQ files in S3 is functionality that can be used elsewhere. Specifically its needed to provide a list of hash counts. This PR does some refactoring and creates a first pass of how the new functionality can be used to count hashes

Test Plan
---------

ensure function still runs and compiles correctly: `python3 -m hmalib.lambdas.pdq.pdq_indexer`

ensure it works correctly in s3:
1.`make upload_docker`
2. delete pdq files and pdq index from s3 using UI
3. Wait for fetcher to rebuild pdq files from TE
4. Ensure pdq_indexer correctly builds index
